src/callbacks_stage/positivity_shallow_water.jl

```diff
@@ -0,0 +1,87 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+"""
+    PositivityPreservingLimiterShallowWater(; variables)
+
+The limiter is specifically designed for the shallow water equations.
+It is applied to all scalar `variables` in their given order
+using the defined `threshold_limiter` from the [`ShallowWaterEquations1D`](@ref) struct
+or the [`ShallowWaterEquations2D`](@ref) struct to determine the minimal acceptable values.
+The order of the `variables` is important and might have a strong influence
+on the robustness.
+
+As opposed to the standard version of the [`PositivityPreservingLimiterZhangShu`](@ref),
+nodes with a water height below the `threshold_limiter` are treated in a special way.
+To avoid numerical problems caused by velocities close to zero,
+the velocity is cut off, such that the node can be identified as "dry". The special feature of the
+`ShallowWaterEquations` used here is that the bottom topography is stored as an additional
```

Comment:
```suggestion
`ShallowWaterEquationsWetDry` used here is that the bottom topography is stored as an additional
```

Reply:
Done in 77f44c650f95e4bc7bcbd284ecd2f60f6b9d0609